### PR TITLE
Mac publishing changes

### LINKED ISSUE #32: Publish for macOS
Currently we publish for GNU/Linux only.

### PR DESCRIPTION
**Pull request by bakshiutkarsha**
should resolve #32

### CODE REVIEW

File: download-libzim.js [hunk 1]
```diff
@@ -56,4 +62,9 @@ for (let url of urls) {
         });
 }
 
-exec(`ln -s lib/x86_64-linux-gnu/libzim.so.6 download/libzim.so.6`);
\ No newline at end of file
+if (os.type() === 'Linux') {
+    exec(`ln -s lib/x86_64-linux-gnu/libzim.so.6 download/libzim.so.6`);
+} else if (os.type() === 'Darwin') {
+    // exec(`ln -s download/lib/libzim.6.dylib /usr/local/lib/libzim.6.dylib`);
+    exec(`cp download/lib/libzim.6.dylib /usr/local/lib`)
```
Comment (bakshiutkarsha) [inline download-libzim.js:69]: I tried creating a symlink for libzim, but I was getting this exception while running test

```
dlopen(/Users/Utkarsha/Documents/Projects/node-libzim/build/Release/zim_binding.node, 1): Library not loaded: libzim.6.dylib
     Referenced from: /Users/Utkarsha/Documents/Projects/node-libzim/build/Release/zim_binding.node
     Reason: image not found
```
So, I directly copied the dynamic library to `/usr/local/lib/libzim.6.dylib`

File: binding.gyp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
     "variables": {
       "libzim_dir": "<(module_root_dir)/download",
+      "libzim_mac_dir": "<(module_root_dir)/download/lib",
```
Comment (kelson42) [inline binding.gyp:4]: I don’t understand why macOS and Linux should have a different directory here!?

File: binding.gyp [hunk 3]
```diff
@@ -28,13 +29,16 @@
...
               }],
               ["OS=='mac'", {
-                "cflags+": ["-fvisibility=hidden"],
-                "xcode_settings": {
-                  "GCC_SYMBOLS_PRIVATE_EXTERN": "YES",  # -fvisibility=hidden
-                  "GCC_ENABLE_CPP_EXCEPTIONS": "YES",
-                },
-                "libraries": [ "-lzim" ]
-              }],
+                    "include_dirs": [
```
Comment (kelson42) [inline binding.gyp:32]: Indentation seems wrong

File: binding.gyp [hunk 4]
```diff
@@ -28,13 +29,16 @@
...
+                        "<(libzim_include)",
+                    ],
+                    "cflags+": ["-fvisibility=hidden"],
+                    "xcode_settings": {
```
Comment (kelson42) [inline binding.gyp:36]: I don’t really understand why this is so different from Linux. Considering, this is the same compiler.

File: download-libzim.js [hunk 5]
```diff
@@ -8,16 +8,22 @@ const urlParser = require('url');
 
 mkdirp.sync('./download');
 
-const isLinux = os.type() === 'Linux';
+const isSupportedOS = os.type() === 'Linux' || os.type() === 'Darwin';
```
Comment (kelson42) [inline download-libzim.js:11]: You should have isLinux and isDarwind instead.

File: download-libzim.js [hunk 6]
```diff
@@ -8,16 +8,22 @@ const urlParser = require('url');
...
+
+let osPrefix = 'linux';
```
Comment (kelson42) [inline download-libzim.js:13]: Please use a ternary operator here.

File: download-libzim.js [hunk 7]
```diff
@@ -56,4 +62,9 @@ for (let url of urls) {
         });
 }
 
-exec(`ln -s lib/x86_64-linux-gnu/libzim.so.6 download/libzim.so.6`);
\ No newline at end of file
+if (os.type() === 'Linux') {
+    exec(`ln -s lib/x86_64-linux-gnu/libzim.so.6 download/libzim.so.6`);
+} else if (os.type() === 'Darwin') {
```
Comment (kelson42) [inline download-libzim.js:67]: This will break if you are not root and is not necessary.

File: download-libzim.js (hunk 1 above)
Comment (kelson42) [reply download-libzim.js:69]: You should not have to do that.

**Comment by kelson42** [comment]
@rgaudin Maybe you have any helpful additional feedback on this PR?

File: download-libzim.js (hunk 7 above)
Comment (kelson42) [reply download-libzim.js:67]: You should have both `libzim.so` and `libzim.so.6`

**Comment by rgaudin** [comment]
I've fixed it but haven't changed the CI. Might be wise to add a macOS target to run the tests there too.

**Pull request merged**